internal/juju/machines.go

```diff
@@ -89,8 +92,51 @@ func (c machinesClient) CreateMachine(ctx context.Context, input *CreateMachineI
 	}
 	defer func() { _ = conn.Close() }()
 
-	machineAPIClient := apimachinemanager.NewClient(conn)
+	response, err := c.createMachine(ctx, conn, input)
+	if err != nil {
+		return nil, err
+	}
+
+	// Wait for machine to go into "running state".
+	retryErr := retry.Call(retry.CallArgs{
+		Func: func() error {
+			modelStatus, err := c.ModelStatus(input.ModelName, conn)
+			if err != nil {
+				return errors.Annotatef(err, "failed to get model status.")
+			}
+			machineStatus, ok := modelStatus.Machines[response.ID]
+			if !ok {
+				return errors.NotFoundf("machine %s", response.ID)
+			}
+			if machineStatus.InstanceStatus.Status == status.Running.String() {
+				return nil
+			}
+			return &keepWaitingError{
+				item:     names.NewMachineTag(response.ID).String(),
+				state:    machineStatus.InstanceStatus.Status,
+				endState: status.Running.String(),
+			}
+		},
+		NotifyFunc: func(err error, attempt int) {
+			if attempt%4 == 0 {
+				message := fmt.Sprintf("waiting for machine %q to be created", response.ID)
+				c.Debugf(message, map[string]interface{}{"attempt": attempt, "err": err})
+			}
+		},
+		IsFatalError: func(err error) bool {
+			return !errors.As(err, &KeepWaitingForError)
+		},
+		BackoffFunc: retry.DoubleDelay,
+		Attempts:    30,
+		Delay:       time.Second,
```

Comment:
question: Why start the retry delay at 1 second if the cache is only updated every 5 seconds?

Reply:
if the cache already has data in it that is  4.1 seconds old, we hit the cache, find the machine not to be running, we retry in 1 sec, cache gets refreshed, we might find the machine running.. i think there are benefits to hitting the cache more often than the refresh interval as long as we don't bombard the controller with model status requests